workflow/executor/executor.go

```diff
@@ -238,6 +243,130 @@ func (we *WorkflowExecutor) LoadArtifacts(ctx context.Context) error {
 	return nil
 }
 
+func (we *WorkflowExecutor) ProcessData(ctx context.Context) error {
+	dataTemplate := we.Template.Data
+	if dataTemplate == nil {
+		return nil
+	}
+
+	// Once we allow input parameters to the data template, we'll load them here
+	var data interface{}
+	var err error
+	for _, step := range dataTemplate {
+		switch {
+		case step.WithArtifactPaths != nil:
+			data, err = we.processDataArtifacts(ctx, step.WithArtifactPaths)
+		case step.Filter != nil:
+			data, err = we.processFilter(data, step.Filter)
+		case step.Aggregator != nil:
+			data, err = we.processAggregator(data, step.Aggregator)
+		}
+
+		if err != nil {
```

Comment:
I'm not a fan of de-coupling error returns like this, I think it's better to always do it immediately after the error is found, this designs-out bugs